WordPress/Classes/Utility/LocationService.m

```diff
@@ -196,18 +177,54 @@ - (void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *
 - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations
 {
     CLLocation *location = [locations lastObject]; // The last item is the most recent.
-
-#if TARGET_IPHONE_SIMULATOR
     [self stopUpdatingLocation];
     [self getAddressForLocation:location];
-#else
-    if (location.horizontalAccuracy > 0 && location.horizontalAccuracy < LocationHorizontalAccuracyThreshold) {
-        [self stopUpdatingLocation];
-        [self getAddressForLocation:location];
-    } else {
-        self.lastUpdatedLocation = location;
+}
+
+- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status
+{
+    if (status == kCLAuthorizationStatusRestricted || status == kCLAuthorizationStatusDenied) {
+        [self serviceFailed:[NSError errorWithDomain:kCLErrorDomain code:kCLErrorDenied userInfo:nil]];
+    }
+}
+
+#pragma mark - Show alert for location errors
+
+- (void)showAlertForLocationServicesDisabled
+{
+    [self showAlertForLocationError:[NSError errorWithDomain:kCLErrorDomain code:kCLErrorDenied userInfo:nil]];
+}
+
+- (void)showAlertForLocationError:(NSError *)error
+{
+    NSString *title = NSLocalizedString(@"Location", @"Title for alert when a generic error happened when trying to find the location of the device");
```

Comment:
Since the title is always `Location`, maybe just define it once here and not redefine it in the if/else block below?